src/database/database-files.cpp

```diff
@@ -396,6 +396,21 @@ bool ModMetadataDatabaseFiles::getModEntries(const std::string &modname, StringM
 	return true;
 }
 
+bool ModMetadataDatabaseFiles::getModKeys(const std::string &modname,
+		std::vector<std::string> *storage)
+{
+	Json::Value *meta = getOrCreateJson(modname);
+	if (!meta)
+		return false;
+
+	std::vector<std::string> keys = meta->getMemberNames();
+	for (std::string &key : keys) {
+		storage->emplace_back(std::move(key));
+	}
```

Comment:
?
```suggestion
	*storage = std::move(keys);
```

Reply:
The database API avoids clearing the given container.

Reply:
```suggestion
	storage->reserve(keys.size());
	for (std::string &key : keys)
		storage->emplace_back(std::move(key));
```
then